frontend/javascripts/oxalis/view/node_context_menu.js

```diff
@@ -0,0 +1,314 @@
+  const isTheSameNode = activeNodeId === nodeContextMenuNodeId;
+  let areNodesConnected = false;
+  if (areInSameTree && !isTheSameNode && activeNodeId != null) {
+    const activeTreeEdges = nodeContextTree.edges.getEdgesForNode(activeNodeId);
+    areNodesConnected = activeTreeEdges.some(
+      edge => edge.source === nodeContextMenuNodeId || edge.target === nodeContextMenuNodeId,
+    );
+  }
+  return (
+    <Menu onClick={hideNodeContextMenu} style={{ borderRadius: 6 }}>
+      <Menu.Item
+        className="node-context-menu-item"
+        key="set-node-active"
+        disabled={isTheSameNode}
+        onClick={() => setActiveNode(nodeContextMenuNodeId)}
+      >
+        Select this Node
+      </Menu.Item>
+      {areInSameTree ? (
+        <Menu.Item
+          className="node-context-menu-item"
+          key="create-edge"
+          disabled={!areInSameTree || isTheSameNode || areNodesConnected}
+          onClick={() =>
+            activeNodeId != null ? createEdge(nodeContextMenuNodeId, activeNodeId) : null
+          }
+        >
+          Create Edge to this Node
+        </Menu.Item>
+      ) : (
+        <Menu.Item
+          className="node-context-menu-item"
+          key="merge-trees"
+          onClick={() =>
+            activeNodeId != null ? mergeTrees(nodeContextMenuNodeId, activeNodeId) : null
+          }
+        >
+          Create Edge & Merge with this Tree
+        </Menu.Item>
+      )}
+      <Menu.Item
+        className="node-context-menu-item"
+        key="delete-edge"
+        disabled={!areNodesConnected}
+        onClick={() =>
+          activeNodeId != null ? deleteEdge(activeNodeId, nodeContextMenuNodeId) : null
+        }
+      >
+        Delete Edge to this Node
+      </Menu.Item>
+      <Menu.Item
+        className="node-context-menu-item"
+        key="delete-node"
+        onClick={() => deleteNode(nodeContextMenuNodeId, nodeContextTree.treeId)}
+      >
+        Delete this Node
+      </Menu.Item>
+      <Menu.Item
+        className="node-context-menu-item"
+        key="measure-node-path-length"
+        disabled={activeNodeId == null || !areInSameTree || isTheSameNode}
+        onClick={() =>
+          activeNodeId != null
+            ? measureAndShowLengthBetweenNodes(activeNodeId, nodeContextMenuNodeId)
+            : null
+        }
+      >
+        Path Length to this Node
+      </Menu.Item>
+      <Menu.Item
+        className="node-context-menu-item"
+        key="measure-whole-tree-length"
+        disabled={activeNodeId == null || !areInSameTree || isTheSameNode}
+        onClick={() => measureAndShowFullTreeLength(nodeContextTree.treeId, nodeContextTree.name)}
+      >
+        Path Length of this Tree
+      </Menu.Item>
+      <Menu.Item
+        className="node-context-menu-item"
+        key="hide-tree"
+        onClick={() => hideTree(nodeContextTree.treeId)}
+      >
+        Hide this Tree
+      </Menu.Item>
+    </Menu>
+  );
+}
+
+function NoneNodeContextMenu({
```

Comment:
Maybe rename to `NoNodeContextMenuOptions` to keep it analoguous to `NodeContextMenuOptions` (also note `No` instead of `None`).